mozilla/fxa
# feat(fxa-settings): Functional ConfirmSignupCode in React

## Because

* We want to make the page created in Storybook fully functional to match parity with the content-server page.

## This pull request

* Activate React route for ConfirmSignupCode.
* Add functionality to verify confirmation code
* Add functionality to resend a new code
* Clean up FormVerifyCode to remove unnecessary props
* Update FormVerifyCode and pages that use it to display an error tooltip when the form is submitted without a code
* Replaces instances of 'security code' with 'authentication code'

## Issue that this pull request solves

Closes #FXA-6493, FXA-6799

## Checklist

_Put an `x` in the boxes that apply_


## Screenshots (Optional)

![image](https://user-images.githubusercontent.com/22231637/221296068-3ff82f59-bb45-44ed-9c24-01803d036344.png)

![image](https://user-images.githubusercontent.com/22231637/221296112-cac32a4e-af65-4b38-ac17-2c45c994bfc0.png)

## Other information (Optional)

### How to test locally

1. Modify `fxa-content-server/server/config/local.json` to enable the showReactApp feature flag for the signUpRoutes (see this doc: https://docs.google.com/document/d/1VZLh7vnUfKsAHNoZla-p7SfYwmwPdmjPUjAeiYe0qfw/edit?usp=sharing).
2. Create a new account on `localhost:3030`
3. At the `/confirm_signup_code` page, add the flag `?showReactApp=true` to view the React page (a quick check to see if you are on the right version - check the dev tools inspector for a `div id='root'`
4. Submitting the form without a code or with a random code should result in error messages.
5. Test a correct code by obtaining it from the CLI with `yarn pm2 logs inbox` - entering that code should redirect to the /settings page with a success alert bar

### Some follow-ups that aren't (yet) handled:

- Handling bounced emails and redirecting
- Using a broker (not yet available in Settings) to redirect based on entrypoint/relying parties
- Using a notifier (not yet available in Settings) to handle newsletter subscription
